apidef/api_definitions.go

```diff
@@ -75,6 +79,76 @@ const (
 	Ignore RoutingTriggerOnType = ""
 )
 
+type ObjectId bson.ObjectId
+
+func (j *ObjectId) Scan(value interface{}) error {
+	var bytes []byte
+	switch v := value.(type) {
+	case []byte:
+		bytes = v
+	case string:
+		bytes = []byte(v)
+	default:
+		return fmt.Errorf("Failed to unmarshal JSON value: %v", value)
+	}
+
+	// reflect magic to update existing string without creating new one
+	if len(bytes) > 0 {
+		bs := ObjectId(bson.ObjectIdHex(string(bytes)))
+		*j = bs
+	}
+
+	return nil
+}
+
+func (j ObjectId) Value() (driver.Value, error) {
+	return bson.ObjectId(j).Hex(), nil
+}
+
+func (j ObjectId) Hex() string {
+	return bson.ObjectId(j).Hex()
+}
+
+func (j ObjectId) Time() time.Time {
+	return bson.ObjectId(j).Time()
+}
+
+func (j ObjectId) Valid() bool {
+	return bson.ObjectId(j).Valid()
+}
+
+func (j ObjectId) String() string {
+	return j.Hex()
+}
+
+func (j ObjectId) GetBSON() (interface{}, error) {
+	return bson.ObjectId(j), nil
+}
+
+func ObjectIdHex(hex string) ObjectId {
+	return ObjectId(bson.ObjectIdHex(hex))
```

Comment:
in case that `hex` is not proper hex format this will panic, should we add prevent of that at this level or?

Reply:
But it is the same as do `bson.ObjectIdHex`. So it really depend on how you handle it on dash side.